.github/workflows/test-identity.yml

```diff
@@ -0,0 +1,46 @@
+name: Run Identity Tests on Identity Code Changes
```

Comment:
This is awesome! Can this be extended to all the domains? As in run secrets test when those are changed? I had created ticket for this https://hashicorp.atlassian.net/browse/HCPF-1722

Reply:
@manish-hashicorp 
Yes! It can be extended manually for sure if each team wants to add their own step to this workflow. If we wanted a more autonomous solution, we might have to spend a bit more time on it.

Roughly speaking, we could do the following:
- Get a listing of all changed files in Git. Parse the list to only show the modified directories.
- Get a recursive listing of all directories within `./internal`
- Loop through the lists and mark which directories to run tests for
- Loop again and actually run the tests with the changed files using `make testacc-ci TEST=DIRECTORY_NAME`

This will be out of scope for this specific ticket since it will require some more testing, but with that being said, I'm happy to update the ticket you posted with my findings. That way, whoever picks it up will have a head start on how to complete the work.

Reply:
Thanks. Please do add information in that ticket.